lib/commonOpts.ts

```diff
@@ -101,11 +102,11 @@ export function getAccountId(
   return getAccountIdFromConfig(portal || account);
 }
 
-export function getMode(options: Arguments<{ mode?: string }>): string {
+export function getMode(options: Arguments<{ mode?: Mode }>): Mode {
   // 1. --mode
   const { mode } = options;
   if (mode && typeof mode === 'string') {
-    return mode.toLowerCase();
+    return mode.toLowerCase() as Mode;
```

Comment:
@joe-yeager, I accidentally included some outdated commits with this PR. I've since rebased to `main`, so going to resolve your comments apart from those on the `errorHandlers` files.